core/src/main/java/org/infinispan/stream/impl/LocalStreamManagerImpl.java

```diff
@@ -247,8 +267,9 @@ public void dataRehashed(DataRehashedEvent<K, V> event) {
       operation.setSupplier(() -> getStream(cacheEntrySet, parallelStream, segments, keysToInclude, keysToExclude));
       operation.handleInjection(registry);
       R value = operation.performOperation();
-      rpc.invokeRemotely(Collections.singleton(origin), factory.buildStreamResponseCommand(requestId, true,
-              Collections.emptySet(), value), rpc.getDefaultRpcOptions(true));
+      CompletableFuture<Map<Address, Response>> completableFuture = rpc.invokeRemotelyAsync(Collections.singleton(origin),
```

Comment:
If `trace` is not on, you could even use `RpcManager.sendTo`. Not sure about the elegance of non-trivial differences in `if (trace) { ... }`

Reply:
Hrmm I never realized that method was there, thanks for the pointer. I should be able to refactor it to a common method that does all the hard work.

Reply:
Ah actually that makes the test behave differently as well as I would have to optionally replace the correct method invocation. I would say for now it can stay as it is to reduce complexity.